server/sandbox_run_linux.go

```diff
@@ -320,20 +320,27 @@ func (s *Server) getSandboxIDMappings(ctx context.Context, sb *libsandbox.Sandbo
 		return nil, errors.New("infra container not found")
 	}
 
-	uids, err := rootless.ReadMappingsProc(fmt.Sprintf("/proc/%d/uid_map", ic.State().Pid))
+	uids, gids, err := unshare.GetHostIDMappings(strconv.Itoa(ic.State().Pid))
```

Comment:
This is the ugly code.

Struct has the exact same shape but it comes from different packages so technically it is a different type. Did a type cast effectively.